web-dev/src/components/AuthenticationLock/index.js

```diff
@@ -0,0 +1,26 @@
+import {Navigate, useLocation} from "react-router-dom";
+import React from "react";
+import {useAuth} from "../../contexts/auth-context";
+
+/**
+ * Put this over component that required authentication to automatically
+ * redirect unauthenticated user to login page
+ * with return URL automatically inserted.
+ *
+ * @param children
+ * @returns {JSX.Element}
+ * @constructor
+ */
+const AuthenticationLock = ({children}) => {
```

Comment:
Is this similar to professor's secure component?

Reply:
Yes, very similar but it adds the returnURL automatically for convenient.